components/export/ui/src/main/resources/PhenoTips/ExportFilter.xml

```diff
@@ -199,17 +199,40 @@
   #set ($query = ", StringProperty externalId $query and externalId.id.id = obj.id and externalId.id.name = 'external_id' and upper(externalId.value) like upper(?)")
   #set ($discard = $parameters.add("$request.get('external_id')%"))
 #end
-#set ($patientIds = $services.query.hql($query).bindValues($parameters))
-#set ($discard = $patientIds.addFilter('unique'))
-#if ($request.count)
-  #set ($discard = $patientIds.addFilter('count'))
-  $patientIds.execute()[0]
-  #break
-#end
-#set ($patientIds = $patientIds.execute())
-#if ($request.list)
-  $response.setContentType('application/json')
-  $jsontool.serialize($patientIds)
+#set ($visibilities = $request.getParameterValues('visibility'))
+#if ("$!visibilities" != '')
+  #set ($query = ",StringProperty visprop, BaseObject visobj $query and visobj.name = doc.fullName and visobj.className = 'PhenoTips.VisibilityClass' and visprop.id.id=visobj.id and visprop.name='visibility' and (")
+  #set ($isFirstVis = true)
```

Comment:
Don't use abbreviations, the few keystrokes you save aren't worth the reduced readability of the code.